### PR TITLE
Pre-processing step for the Python interface

### PR DESCRIPTION
**Pull request by hariszaf**
If applied, this commit will support the pre-processing step for the Python interface
The ```pre_process()``` function is now available and enables this task
The function is located in the ```volestipy.pyx``` however it is purely Python

### CODE REVIEW

File: volestipy/volestipy/volestipy.pyx [hunk 1]
```diff
@@ -225,7 +345,7 @@ cdef class low_dim_HPolytope:
       return np.asarray(self._Aeq)
    @property
    def beq(self):
-      return np.asarray(self._beq)   
+      return np.asarray(self._beq)
    @property
    def dimensions(self):
-      return self._A.shape[1]   
\ No newline at end of file
+      return self._A.shape[1]
```
Comment (TolisChal) [inline volestipy/volestipy/volestipy.pyx:570]: please add an empty line at the end of all files that contain code in folder `volestipy`

File: volestipy/volestipy/volestipy.pyx [hunk 2]
```diff
@@ -3,26 +3,146 @@
...
+
+         # Optimize model
+         model.optimize ()
+
+         # If optimized, print the solution
+         status = model.status
+         if status == GRB.OPTIMAL:
+            solution = model.getAttr('x')
+            print("\n\n***The solution for the MAX case is:")
+            print(solution)
```
Comment (TolisChal) [inline volestipy/volestipy/volestipy.pyx]: Do we need printing here? I think printing should be used only in tests

File: volestipy/volestipy/volestipy.pyx [hunk 3]
```diff
@@ -3,26 +3,146 @@
...
+         model.update()
+         model.optimize()
+
+         # Again if optimized, print the solution
+         status = model.status
+         if status == GRB.OPTIMAL:
+            solution = model.getAttr('x')
+
+            print("\n\n***The solution for the MIN case is:")
+            print(solution)
```
Comment (TolisChal) [inline volestipy/volestipy/volestipy.pyx]: Please remove printing

File: volestipy/volestipy/volestipy.pyx (hunk 1 above)
Comment (hariszaf) [reply volestipy/volestipy/volestipy.pyx:570]: done.

File: volestipy/volestipy/volestipy.pyx (hunk 2 above)
Comment (hariszaf) [reply volestipy/volestipy/volestipy.pyx]: Nope. I removed the ```solution``` variable completely

File: volestipy/volestipy/volestipy.pyx (hunk 3 above)
Comment (hariszaf) [reply volestipy/volestipy/volestipy.pyx]: Likewise

File: volestipy/README.md [hunk 4]
```diff
@@ -1,28 +1,70 @@
...
+
+### Install *volestipy*
+
+After getting the dependencies, download the *volesti* repository from GitHub and run the following commands.
 
 ```
 cd volestipy
-LDFLAGS="-L/usr/lib/lpsolve/" python setup.py build_ext -i
-python setup.py install -f
+LDFLAGS="-L/usr/lib/lp_solve/" python3 setup.py install --user
```
Comment (vissarion) [inline volestipy/README.md]: you both use `lp_solve` and 'lpsolve` in README, please be consistent

File: volestipy/tests/bigg_files/e_coli_core.json [hunk 5]
```diff
@@ -0,0 +1,14786 @@
+{
```
Comment (vissarion) [inline volestipy/tests/bigg_files/e_coli_core.json]: this is extremely huge for inclusion in the library, if it is not publicly available I recommend to create another repository or a dataset (e.g. in zenodo) and put it there

File: volestipy/volestipy/volestipy.pyx (hunk 1 above)
Comment (vissarion) [reply volestipy/volestipy/volestipy.pyx:570]: these are four lines, one is enough ;)

File: volestipy/tests/bigg_files/e_coli_core.json (hunk 5 above)
Comment (vissarion) [reply volestipy/tests/bigg_files/e_coli_core.json]: EDIT: it is available here: http://bigg.ucsd.edu/models/e_coli_core so you have to retrieve it from there

**Review by vissarion — COMMENTED** [review]
Nice addition, thanks! 

My main comment is about large files (i.e. `json`) and binary (i.e. `.mat`). There should not be in the repo. They are available in the web e.g. http://bigg.ucsd.edu/models/e_coli_core

**Comment by vissarion** [comment]
I am trying to run the tests locally. In `volestipy` I am running `python3 tests/test_rounding.py` and get 
```The polytope:
Dimensions: 5
A matrix:
[[ 1.  0.  0.  0.  0.]
 [ 0.  1.  0.  0.  0.]
 [ 0.  0.  1.  0.  0.]
 [ 0.  0.  0.  1.  0.]
 [ 0.  0.  0.  0.  1.]
 [-1.  0.  0.  0.  0.]
 [ 0. -1.  0.  0.  0.]
 [ 0.  0. -1.  0.  0.]
 [ 0.  0.  0. -1.  0.]
 [ 0.  0.  0.  0. -1.]]
<class 'numpy.ndarray'>
b vector:
[ 1.  1.  1.  1.  1.  1.  1.  1.  1.  1.]
<class 'numpy.ndarray'>
*** This is a test for th rounding() function of the volestipy library *** 

Traceback (most recent call last):
  File "tests/test_rounding.py", line 31, in <module>
    rounding_output_max_ellipsoid = hp.rounding(rounding_method = "max_ellipsoid")
  File "volestipy/volestipy.pyx", line 438, in volestipy.HPolytope.rounding
    np.savetxt('round_value.txt', np.asarray(round_value))
  File "/usr/lib/python3/dist-packages/numpy/lib/npyio.py", line 1214, in savetxt
    ncol = X.shape[1]
IndexError: tuple index out of range
```
The rest of the tests seems to run OK.

File: volestipy/tests/bigg_files/e_coli_core.json (hunk 5 above)
Comment (hariszaf) [reply volestipy/tests/bigg_files/e_coli_core.json]: Files removed from the repo.

File: volestipy/README.md (hunk 4 above)
Comment (hariszaf) [reply volestipy/README.md]: replcaced ```lp_solve``` with ```lpsolve```

**Review by TolisChal — APPROVED** [review]

File: volestipy/volestipy/volestipy.pyx (hunk 1 above)
Comment (vissarion) [reply volestipy/volestipy/volestipy.pyx:570]: hmm, you are back to the empty line issue now ;)

**Review by vissarion — APPROVED** [review]
Thanks for the revision. If you remove the empty  line issue, I am OK with merging.

File: volestipy/volestipy/volestipy.pyx (hunk 1 above)
Comment (hariszaf) [reply volestipy/volestipy/volestipy.pyx:570]: There is something strange with my editor I think..
If I am not mistaken, last line in the ```volestipy.pyx``` file is number 571 and it is empty..
Is it not?

**Pull request merged**